pkg/backend/filestate/store.go

```diff
@@ -232,6 +232,22 @@ func (p *projectReferenceStore) ListProjects(ctx context.Context) ([]tokens.Name
 	return projects, nil
 }
 
+func (p *projectReferenceStore) ProjectExists(ctx context.Context, projectName string) (bool, error) {
+	if projectName == "" {
+		return false, fmt.Errorf("empty project name")
+	}
+
+	path := filepath.Join(StacksDir, projectName)
+
+	files, err := listBucket(ctx, p.bucket, path)
+	if err != nil {
+		return false, fmt.Errorf("error listing stacks: %w", err)
```

Comment:
minor thing: When chaining errors in Go, we don't need to add "error doing X" or "failed to do Y". If every node in the chain did that, then we'd end up with fairly verbose error messages. we can change this to something like:

```suggestion
		return false, fmt.Errorf("list stacks at %q: %w", path, err)
```

Related resources if you're curious to understand the rationale further.

- https://github.com/uber-go/guide/blob/master/style.md#error-wrapping
- https://preslav.me/2023/04/14/golang-error-handling-is-a-form-of-storytelling/

Reply:
OK. Thanks for the references.

Reply:
Updated https://github.com/pulumi/pulumi/pull/12798/commits/48756f68b1ec77b93f1ada9bc9dba98a5d1ad806